src/richie/apps/search/filter_definitions/base.py

```diff
@@ -207,12 +210,30 @@ def get_faceted_definitions(self, facets, *args, **kwargs):
 
         # for each facet, we derive the value and the count:
         # eg. for facet key `availability@coming_soon`, the value is `coming_soon`
-        facet_counts_dict = {
-            key.split("@")[1]: facet["doc_count"]
+        facet_counts = [
+            (key.split("@")[1], facet["doc_count"])
             for key, facet in facets.items()
             if "{:s}@".format(self.name) in key  # 6 times faster than startswith
             and facet["doc_count"] >= self.min_doc_count
-        }
```

Comment:
In fact, we have a bug here (already existing) on a edge case:

> if a selected facet has a count of less than `min_doc_count` it will be left out and shouldn't

I think we can solve the bug and simplify the code below by working on the condition in the dict comprehension. I didn't test it but maybe something like:
```python
facet_name = "{:s}@".format(self.name)
sorted_facets = sorted(facets.items(), key=lambda f: f[1]["doc_count'])
facet_counts_dict = {
    key.split("@")[1]: facet["doc_count"]
    for i, (key, facet) in enumerate(sorted_facets)
    if facet_name in key  # 6 times faster than startswith 
    and key in data[self.name]
    or facet["doc_count"] >= self.min_doc_count and  i < facet_counts_limit  # or i <= maybe?
}
has_more_values = len(facet_counts_dict) > facet_counts_limit